lib/PipelineC/CMakeLists.txt

```diff
@@ -2,6 +2,34 @@
 # This file is distributed under the MIT License. See LICENSE.md for details.
 #
 
-revng_add_library_internal(revngPipelineC SHARED PipelineC.cpp)
+add_custom_command(
+  OUTPUT "${CMAKE_BINARY_DIR}/lib/PipelineC/types.inc"
+         "${CMAKE_BINARY_DIR}/lib/PipelineC/functions.inc"
+         "${CMAKE_BINARY_DIR}/lib/PipelineC/wrappers.inc"
+  COMMAND
+    # cmake-format: off
+    python "${CMAKE_SOURCE_DIR}/scripts/PipelineC_add_tracing.py"
+    -i "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/ForwardDeclarationsC.h"
+    -i "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/Prototypes.h"
+    "${CMAKE_BINARY_DIR}/lib/PipelineC"
+    # cmake-format: on
+  DEPENDS "${CMAKE_SOURCE_DIR}/scripts/PipelineC_add_tracing.py"
+          "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/ForwardDeclarationsC.h"
+          "${CMAKE_SOURCE_DIR}/include/revng/PipelineC/Prototypes.h")
+
+revng_add_library_internal(
+  revngPipelineC SHARED "${CMAKE_BINARY_DIR}/lib/PipelineC/types.inc"
+  "${CMAKE_BINARY_DIR}/lib/PipelineC/wrappers.inc" Impl/PipelineC.cpp)
+
+revng_add_library_internal(
+  revngPipelineCTracing "${CMAKE_BINARY_DIR}/lib/PipelineC/types.inc"
+  "${CMAKE_BINARY_DIR}/lib/PipelineC/functions.inc" Tracing/Inspector.cpp
+  Tracing/Runner.cpp)
+
+target_include_directories(revngPipelineC
+                           PRIVATE "${CMAKE_BINARY_DIR}/lib/PipelineC")
+target_include_directories(revngPipelineCTracing
+                           PRIVATE "${CMAKE_BINARY_DIR}/lib/PipelineC")
```

Comment:
We already have `$BUILD_DIR/include` in the list of include paths. Generate the files in `$BUILD_DIR/include/revng/PipelineC/...` as appropriate.